src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -112,45 +113,54 @@ class RankBasedMetricResults(MetricResults):
     """
 
     #: The mean over all ranks: mean_i r_i. Lower is better.
-    mean_rank: Dict[str, float] = field(metadata=dict(doc='The mean over all ranks: mean_i r_i. Lower is better.'))
+    mean_rank: Dict[str, Dict[str, float]] = field(metadata=dict(
+        doc='The mean over all ranks: mean_i r_i. Lower is better.',
+    ))
 
     #: The mean over all reciprocal ranks: mean_i (1/r_i). Higher is better.
-    mean_reciprocal_rank: Dict[str, float] = field(metadata=dict(
+    mean_reciprocal_rank: Dict[str, Dict[str, float]] = field(metadata=dict(
         doc='The mean over all reciprocal ranks: mean_i (1/r_i). Higher is better.',
     ))
 
     #: The hits at k for different values of k, i.e. the relative frequency of ranks not larger than k.
     #: Higher is better.
-    hits_at_k: Dict[str, Dict[int, float]] = field(metadata=dict(
+    hits_at_k: Dict[str, Dict[str, Dict[int, float]]] = field(metadata=dict(
         doc='The hits at k for different values of k, i.e. the relative frequency of ranks not larger than k.'
             ' Higher is better.',
     ))
 
     #: The mean over all chance-adjusted ranks: mean_i (2r_i / (num_entities+1)). Lower is better.
     #: Described by [berrendorf2020]_.
-    adjusted_mean_rank: float = field(metadata=dict(
+    adjusted_mean_rank: Dict[str, float] = field(metadata=dict(
         doc='The mean over all chance-adjusted ranks: mean_i (2r_i / (num_entities+1)). Lower is better.',
     ))
 
     def get_metric(self, name: str) -> float:  # noqa: D102
         if name == 'adjusted_mean_rank':
-            return self.adjusted_mean_rank
+            # TODO: Sided adjusted mean rank
```

Comment:
@cthoyt Any ideas how to fix this?

The basic problem is that we want to have default side `both` and default rank_type `avg`, but need to determine which one was provided for e.g. `avg.mean_rank`.

Reply:
Is this still a problem? From the most recent code it looks like you made the same fix I would have proposed

Reply:
Yes. For `head.adjusted_mean_rank` we have to provide `head.avg.adjusted_mean_rank`, although `avg` is the only admissible `rank_type` for `adjusted_mean_rank`. Just writing `head.adjusted_mean_rank` will let the code assume that we wanted to use `rank_type="head"` on both sides.

Reply:
I have pushed an example. The basic idea is that based on the logic that `head.adjusted_mean_rank` will use `rank_type='avg'`, since it is the only one allowed, all other metrics will use the default, which is `'avg'`, such that `head.mean_rank` is also possible. The code will check whether the user mean a side or rank_type when just one dot was provided and will always ensure that `rank_type=='avg'` for the 'adjusted_mean_rank` metric.

Reply:
Looks good to me :+1: